files/build_templates/docker_image_ctl.j2

```diff
@@ -315,6 +315,35 @@ start() {
         source $PLATFORM_ENV_CONF
     fi
 
+    {%- if sonic_asic_platform == "broadcom" %}
+    {%- if docker_container_name == "syncd" %}
+    # Set the SYNCD_SHM_SIZE if this variable not defined
+    SYNCD_SHM_INI=/usr/share/sonic/device/profiles/syncd_shm.ini
+    BRCM_PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+
+    readline=$(grep '0x14e4' /proc/linux-kernel-bde)
+    bcm_chip_id=${readline#*0x14e4:0x}
+    bcm_chip_id=${chip_id::3}
+    if [ -f $BRCM_PLATFORM_COMMON_DIR/x86_64-broadcom_${bcm_chip_id}/broadcom-sonic*.config.bcm ]; then
```

Comment:
the logic here sounds complicated, why not map the shm_size directly to ${bcm_chip_id}?

Reply:
cause the chip id is not well-known Ids? they are like b77(for name td3), b85(for name td2) ...
Then, for a better known, map it to the name

Reply:
simplified.